core/src/main/java/google/registry/flows/domain/DomainPricingLogic.java

```diff
@@ -122,6 +128,64 @@ FeesAndCredits getRenewPrice(Registry registry, String domainName, DateTime date
             .build());
   }
 
+  /** Returns a new renewal cost for the pricer. */
+  @SuppressWarnings("unused")
+  FeesAndCredits getRenewPrice(
+      Registry registry,
+      String domainName,
+      DateTime dateTime,
+      int years,
+      @Nullable Recurring recurringBillingEvent)
+      throws EppException {
+    checkArgument(years > 0, "Number of years must be positive");
+    Money renewCost = PricingEngineProxy.getDomainRenewCost(domainName, dateTime, years);
```

Comment:
Don't initialize this here. It's confusing, as it's always being overwritten later on (by even the same thing in the `DEFAULT` case, so you're calling the pricing proxy twice for no reason on that path).

Reply:
Yeah, I think you should only be calling this in the else.  So it'd look something like:

```
Money renewCost;
if (recurringBillingEvent == null) {
  renewCost = PricingEngineProxy.getDomainRenewCost(domainName, dateTime, years);
} else {
  switch(...) {
    ......
  }
}
```